### PR TITLE
Add ASAN test, first drop

### PR DESCRIPTION
**Pull request by leshikus**
The container still needs to be updated with new versions of LLVM, parquet, etc

### CODE REVIEW

File: .github/workflows/test-docker.yml [hunk 1]
```diff
@@ -47,8 +50,15 @@ jobs:
           docker exec hdk-build.${{ inputs.name }} chown -R ghrunner:ghrunner /_work/
           docker exec -u ghrunner hdk-build.${{ inputs.name }} tar -zxf /_work/build.tgz -C /_work/
 
-      - name: Test
-        run: docker exec -u ghrunner hdk-build.${{ inputs.name }} bash /_work/omniscidb/scripts/conda/test.sh
+      - name: Smoke test
+        if: inputs.name == 'cuda'
```
Comment (kurapov-peter) [inline .github/workflows/test-docker.yml:54]: Is this a cuda-only flow?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -68,6 +68,32 @@ jobs:
...
+
+        run: |
+          sudo sh -c 'echo 2 >/proc/sys/vm/overcommit_memory'
+          cd omniscidb
+
+          cd build/Tests # build cache
+          conda run -n ${{ env.CONDA_ENV }} ./ArrowBasedExecuteTest --build-rel-alg-cache=cache.txt
+
+          cd ../.. # rebuild with ASAN
+          conda run -n ${{ env.CONDA_ENV }} cmake -B ./build -S . -DENABLE_TESTS=on -DENABLE_CUDA=off -DENABLE_ASAN=on
```
Comment (kurapov-peter) [inline .github/workflows/test.yml:84]: Is this for saving up space or?

**Review by kurapov-peter — COMMENTED** [review]
Looks fine, a couple questions.

File: .github/workflows/test-docker.yml (hunk 1 above)
Comment (leshikus) [reply .github/workflows/test-docker.yml:54]: this is a "self hosted" workflow; it applies now to two scenarios: CUDA tests and ASAN tests e g those we cannot run with github runners

Comment (leshikus) [reply .github/workflows/test-docker.yml:54]: the strange way to pass "cuda" into a reusable workflow and query for it comes from two requirements/limitations:
1) Igor requests no cut&paste;
2) it was not possible (until recently) to call a function from another function

this results in the following approach

<common code>
<code for a cuda runner> if cuda
<code for an asan runner> if asan

File: .github/workflows/test.yml (hunk 2 above)
Comment (leshikus) [reply .github/workflows/test.yml:84]: we need to run an unmodified test first to cache calcite queries and responses, this is done in this patch
then we need to recompile the whole project with ASAN and run the test with cached queries/responses, this is not fully finished

Comment (kurapov-peter) [reply .github/workflows/test.yml:84]: Ah, I see

**Review by kurapov-peter — APPROVED** [review]
LGTM

**Pull request merged**